modules/telestion-api/src/main/java/de/wuespace/telestion/api/message/header/Information.java

```diff
@@ -0,0 +1,164 @@
+					.formatted(this.getClass().getName()));
+			return headers;
+		}
+
+		for (var component : this.getClass().getRecordComponents()) {
+			try {
+				var serialization = SerializationUtils.serialize(component);
+				var value = getOrDefault(component);
+				headers.set(serialization.name(), InformationCodec.encode(value));
+			} catch (NoSerializationException e) {
+				logger.warn(e.getMessage(), e);
+			} catch (IllegalAccessException | InvocationTargetException e) {
+				logger.error("An unexpected exception occurred. Cannot access record value {} in {}. " +
+								"Please file a bug report at https://github.com/wuespace/telestion-core/issues/new",
+						component.getName(), this.getClass().getName(), e);
+			}
+		}
+
+		return headers;
+	}
+
+	/**
+	 * Like {@link #appendToHeaders(MultiMap)} but uses new and empty headers instead.
+	 *
+	 * @return new headers only with the packed record information
+	 */
+	default MultiMap toHeaders() {
+		return appendToHeaders(MultiMap.caseInsensitiveMultiMap());
+	}
+
+	/**
+	 * Extracts record information from Vert.x message headers based on the serialization information
+	 * on each record component.
+	 * <p>
+	 * If a record value is {@code null}, the default value from serialization information is used instead.
+	 *
+	 * @param headers          Vert.x message headers containing the required information
+	 *                         to construct an information record
+	 * @param informationClass the type of the information record
+	 * @return the constructed information record or {@code null} if no value
+	 * for a non-nullable record component is found
+	 */
+	static <T> T fromHeaders(MultiMap headers, Class<T> informationClass) {
+		if (!informationClass.isRecord()) {
+			logger.error("%s is not a record. Please use a record as information type to use header utility functions."
+					.formatted(informationClass.getName()));
+			return null;
+		}
+
+		var values = new ArrayList<>();
+
+		for (var component : informationClass.getRecordComponents()) {
+			try {
+				var serialization = SerializationUtils.serialize(component);
+				var defaultValue = SerializationUtils.nullableDefaultValue(serialization);
+				var value = headers.get(serialization.name());
+				value = Objects.isNull(value) ? defaultValue : value;
+
+				if (Objects.isNull(value)) {
+					if (!serialization.isNullable()) {
+						logger.error("No value for record component {} found, but is explicitly required.",
+								component.getName());
+						return null;
+					}
+					values.add(InformationCodec.getDefault(component.getType()));
+					continue;
+				}
+
+				values.add(serialization.isUnsigned()
+						? InformationCodec.decodeUnsigned(value, component.getType())
+						: InformationCodec.decode(value, component.getType()));
+			} catch (NoSerializationException | NoSuchPrimitiveTypeException e) {
+				logger.warn(e.getMessage(), e);
+				values.add(null);
```

Comment:
I have the bad feeling that this will bring you hard to debug NPEs at a later point.

Reply:
Could you elaborate on this a little bit more?